dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java

```diff
@@ -1845,17 +1844,15 @@ private void prepareCaches( User user, List<Event> events )
     {
         if ( params.getOrgUnit() != null )
         {
-            return Arrays.asList( params.getOrgUnit() );
+            return Collections.emptyList();
```

Comment:
Just curious about the logic here?

Reply:
the logic is we have two blocks of orgunits that the query generator will use - param.orgunit and/or those obtained through ouMode. 

We always prioritize param.orgunit. 

In this case user has both ouMode=ALL and param.orgunit. so we are saying orgunits obtained through ouMode is blank and the query will use param.orgunit, we are returning immediately without even checking whether use can search across all. 

If param.orguint is null, then we check if user can search across all. If user can search across all we return empty also ... but this time it is different because both param.orgunit and the ouMode orgunit are blank ... then the query goes without orgunit in the where clause.